### PR TITLE
Rename schemas part 1 #508

### PR DESCRIPTION
**Pull request by saraswatpuneet**
# Goal
The goal of this PR is
- [x] Schemas `SchemaCount`, `schema_count`
  - `CurrentSchemaIdentifierMaximum`, `get_current_schema_identifier_maximum`
- [x] Schemas `register_schema`
  - Schemas `create_schema`?
- [x] https://github.com/LibertyDSNP/frequency/pull/545
  - `msa_grantedSchemaIdsByMsaId`?
 
Part of #508 

# Checklist
- [x] n/a Chain spec updated
- [x] n/a Updated the js/api-augment code if a custom RPC added/changed
- [x] Design doc(s) updated
- [x] n/a Tests added
- [x] Benchmarks updated
- [x] n/a Weights updated

### CODE REVIEW

File: pallets/schemas/src/lib.rs [hunk 1]
```diff
@@ -258,20 +259,20 @@ pub mod pallet {
...
 
 		fn add_schema(
 			model: BoundedVec<u8, T::SchemaModelMaxBytesBoundedVecLimit>,
 			model_type: ModelType,
 			payload_location: PayloadLocation,
 		) -> Result<SchemaId, DispatchError> {
-			let cur_count = Self::get_schema_count();
+			let cur_count = Self::get_current_schema_identifier_maximum();
 			ensure!(cur_count < T::MaxSchemaRegistrations::get(), Error::<T>::TooManySchemas);
 			let schema_id = cur_count.checked_add(1).ok_or(Error::<T>::SchemaCountOverflow)?;
```
Comment (dopatraman) [inline pallets/schemas/src/lib.rs:272]: Should we encapsulate this logic in a method similar to the MSA pallet?
```
	pub fn get_next_msa_id() -> Result<MessageSourceId, DispatchError> {
		let next = Self::get_identifier().checked_add(1).ok_or(Error::<T>::MsaIdOverflow)?;
		let next = Self::get_current_msa_identifier_maximum()
			.checked_add(1)
			.ok_or(Error::<T>::MsaIdOverflow)?;

		Ok(next)
	}

```

Comment (saraswatpuneet) [reply pallets/schemas/src/lib.rs:272]: I think this is different and safe way to do count+1

In the one you commented, that is getting the total count while adding which is bound to be the number current schema is being added again

Comment (saraswatpuneet) [reply pallets/schemas/src/lib.rs:272]: I see what you mean having looked at both functions


Good suggestion let me do it

**Review by shannonwells — APPROVED** [review]
Let's roll

**Pull request merged**